### DISCUSSION TITLE
Released stages dependent WI in output WIKI article

### DISCUSSION THREAD

**Discussion by ArtemNaydenko** [rfennell/AzurePipelines#1123] [Q&A]
Hey, @rfennell!

Firstly, I would like to thank you. Your plugin is simply a life savior, thanks for your hard work. 
I am using Generate Release Notes (Crossplatform) + WIKI Updater Tasks bond. It works as expected, returns the difference between the last successful release and the current one. Then it posts notes in a defined section on project wiki. But I want to add a little to the top and somehow to add a stage check. E.g.: I have multiple stages for each build and I can release them separately. Is there any way to define the stage within the template and append release notes article on wiki with only deployed stage related WIs? 

Played a little with currentStage object. I tried to dump it to receive it's params: # currentStage {{json currentStage}}. But it returned nothing:
![image](https://user-images.githubusercontent.com/93381791/139427108-4418ea35-6470-4e90-aa33-210ae42648a7.png)

The next step I was thinking about, was adding some kind of #if rule to check if the stage has been actually released and to dump only associated WIs. But not sure about this as well. 

Thank you very much in advance!

**Comment by rfennell** [comment]
A few things might help

- If you set the parameter `checkstage=true` on the task it will do the comparison against the last time there was a release to that stage. This can give stage-specific release notes. This parameter has to be set to populate the `currentStage` template object
- You can used the `ReplaceFile` and `AppendFile` task parameters to define how the release notes are built

| ReplaceFile | AppendFix | What do you get 
| True            | Ignored     | You always get a new file 
| False           | True           | The new content is appended to the existing file, if no file exists a new one is created |
| False           | False          | The new content is prepended to the existing file, if no file exists a new one is created |

Hope that helps
